lighthouse-core/report/v2/report-renderer.js

```diff
@@ -181,6 +182,36 @@ class ReportRenderer {
   }
 
   /**
+   * @param {string} text
+   * @return {!Element}
+   */
+  _convertMarkdownLinksToElement(text) {
+    const element = this._createElement('span');
+
+    // Split on markdown links (e.g. [some link](https://...)).
+    const parts = text.split(/(\[(.*?)\]\((https?:\/\/.*?)\))/g);
+
+    while (parts.length) {
+      // Remove the same number of elements as there are capture groups.
```

Comment:
you could also avoid the `splice` by doing something like
```js
let parts = text.split(/(\[(.*?)\]\((https?:\/\/.*?)\))/g);
while (parts.length) {
  const [preambleText, fullMarkdownLink, linkText, linkHref, ...rest] = parts;
  // ...
  parts = rest;
}
```
more explicit, though the extra assignment is kind of annoying